flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -313,7 +329,14 @@ private void flushIfAble() {
      */
     private void flush() {
         while (inFlightRequestsCount >= maxInFlightRequests) {
-            mailboxExecutor.tryYield();
+            try {
+                mailboxExecutor.yield();
+            } catch (InterruptedException e) {
```

Comment:
Why do you handle the InterruptedException here? I think it is better to add it to the method signature because `write` allows throwing and `InterruptedException`

Reply:
Makes sense, it was a tough decision for me to handle it here or in the 4 places that call `flush()`. 3 of those 4 places throw an `InterruptedException` executing on the mailbox thread anyway, so it has the same effect. So I will move this block to the fourth and final place the `flush(boolean flush)` method.

Reply:
Implemented! :-)

Reply:
I think you do not need to handle the `InterruptedException` at all. The `flush(boolean end)` method allows throwing the exception.